File: backend/src/main/java/com/management/employee/service/EmployeeService.java
package com.management.employee.service;

import com.management.employee.model.Employee;
import com.management.employee.repository.EmployeeRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmployeeService {

    @Autowired
    EmployeeRepo employeeRepo;

    public List<Employee> getAllEmployees() {
        return employeeRepo.findAll();
    }

    public boolean addEmployee(Long empId, String empName, String designation, String empEmail, String workLocation) {
        // Check if the employee already exists
        if (employeeRepo.existsById(empId)) {
            return false; // Employee already exists
        }

        // Create a new employee object
        Employee employee = new Employee(empId, empName, designation, empEmail, workLocation);


        // Save the new employee to the repository
        employeeRepo.save(employee);
        return true; // Employee added successfully
    }

    public boolean updateEmployee(Long empId, String empName, String designation, String empEmail, String workLocation) {
        // Check if the employee exists
        if (!employeeRepo.existsById(empId)) {
            return false; // Employee does not exist
        }

        // Create a new employee object with updated details
        Employee employee = new Employee(empId, empName, designation, empEmail, workLocation);

        // Save the updated employee to the repository
        employeeRepo.save(employee);
        return true; // Employee updated successfully
    }

    public boolean deleteEmployee(Long empId) {
        // Check if the employee exists
        if (!employeeRepo.existsById(empId)) {
            return false; // Employee does not exist
        }

        // Delete the employee from the repository
        employeeRepo.deleteById(empId);
        return true; // Employee deleted successfully
    }

    public Employee getEmployeeById(Long empId) {
        return employeeRepo.findByEmpId(empId);
    }
}

File: frontend/ems-frontend/src/services/EmployeeService.js
import axios from "axios";

const API_BASE_URL = "http://54.209.178.0:8080/employees";
export const listEmployees = axios.get(API_BASE_URL);

export const getEmployeeById = (empId) => {
  return axios.get(`${API_BASE_URL}/${empId}`);
};

export const addEmployee = (employee) => {
  return axios.post(`${API_BASE_URL}/add`, employee);
};

export const updateEmployee = (employee) => {
  return axios.put(`${API_BASE_URL}/update`, employee);
};

export const deleteEmployee = (empId) => {
  return axios.delete(`${API_BASE_URL}/delete/${empId}`);
};
